speechbrain/k2_integration/utils.py

```diff
@@ -0,0 +1,162 @@
+        aux_shape = aux_shape.remove_axis(1)
+        aux_shape = aux_shape.remove_axis(1)
+        aux_labels = k2.RaggedTensor(aux_shape, aux_labels.values)
+    else:
+        # remove axis corresponding to states.
+        aux_shape = best_paths.arcs.shape().remove_axis(1)
+        aux_labels = k2.RaggedTensor(aux_shape, best_paths.aux_labels)
+        # remove 0's and -1's.
+        aux_labels = aux_labels.remove_values_leq(0)
+
+    assert aux_labels.num_axes == 2
+    if return_ragged:
+        return aux_labels
+    else:
+        return aux_labels.tolist()
+
+
+def lattice_paths_to_text(best_paths: k2.Fsa, word_table) -> List[str]:
+    """
+    Convert the best path to a list of strings.
+
+    Arguments
+    ---------
+    best_paths: k2.Fsa
+        It is the path in the lattice with the highest score for a
+        given utterance.
+
+    Returns
+    -------
+    texts: List[str]
+        A list of strings, each of which is the decoding result of the
+        corresponding utterance.
+    """
+    hyps: List[List[int]] = lattice_path_to_textid(
+        best_paths, return_ragged=False
+    )
+    texts = []
+    for wids in hyps:
+        texts.append(" ".join([word_table[wid] for wid in wids]))
+    return texts
+
+
+def load_G(path: Union[str, Path], cache: bool = True) -> k2.Fsa:
+    """
+    load a lm to be used in the decoding graph creation (or lm rescoring).
+
+    Arguments
+    ---------
+    path: str
+        The path to an FST LM (ending with .fst.txt) or a k2-converted
+        LM (in pytorch .pt format).
+    cache: bool
+        Whether or not to load/cache the LM from/to the .pt format (in the same dir).
+
+    Returns
+    -------
+    G: k2.Fsa
+        An FSA representing the LM.
+    """
+    path = str(path)
+    if os.path.exists(path.replace(".fst.txt", ".pt")) and cache:
+        logger.warning(
+            f"Loading '{path}' from its cached .pt format."
+            " Set 'caching: False' in the yaml"
+            " if this is not what you want."
+        )
+        G = k2.Fsa.from_dict(
+            torch.load(path.replace(".fst.txt", ".pt"), map_location="cpu")
+        )
+        return G
+
+    logger.info(f"Loading G LM: {path}")
+    # If G_path is an fst.txt file then convert to .pt file
+    if not os.path.isfile(path):
+        raise FileNotFoundError(
+            f"File {path} not found. " "You need to run arpa_to_fst to get it."
+        )
+    with open(path) as f:
+        G = k2.Fsa.from_openfst(f.read(), acceptor=False)
+        torch.save(G.as_dict(), path[:-8] + ".pt")
+    return G
+
+
+def prepare_rescoring_G(G: k2.Fsa) -> k2.Fsa:
+    """
+    Prepare a LM with the purpose of using it for LM rescoring.
+    For instance, in the librispeech recipe this is a 4-gram LM (while a
+    3gram LM is used for HLG construction).
+
+    Arguments
+    ---------
+    path: str
+        The path to an FST LM (ending with .fst.txt) or a k2-converted
+        LM (in pytorch .pt format).
```

Comment:
G should be there and you should remove path as it is not part of the arguments of this function

Reply:
Solved.